### DISCUSSION TITLE
Schema.registerTypeDefaultComponents方法无效？

### DISCUSSION THREAD

**Discussion by yuyi919** [alibaba/formily#1622] [Q&A 2.x]
https://codesandbox.io/s/delicate-frog-x6jcz?file=/src/App.vue

这个sandbox里总共发现了两个问题
1.Schema.registerTypeDefaultComponents无效，于是我使用registerPatches实现类似效果
2.registerPatches似乎是在处理$ref之前处理的，而且没有处理到definitions

三个建议
1.是否可以添加Schema.registerTypeDefaultDecorator方法？可以定义默认的decorator
2.registerPathes可以处理到definitions下的定义
3.看文档描述triggerType需要单独定义，可否针对组件也提供一个默认的triggerType的配置
   比如input组件可以单独定义默认在onBlur时校验，其它组件不提供配置默认还是在onChange时校验

**Comment by yuyi919** [comment]
另外尝试发现，triggerType只允许传入一种类型，能否允许传入一个数组，如["onChange", "onBlur"]

**Reply by janryWang** [reply]
onChange本身就是所有情况都会触发，你传这个数组没意义

**Reply by yuyi919** [reply]
这可能是一种习惯问题吧，我们这边的习惯是inpu组件没录入就blur也会触发 :)

**Comment by janryWang** [comment]
registerTypeDefaultComponents 要求必须enablePolyfill('1.0')

**Reply by yuyi919** [reply]
原来如此，我在@formily/react下找到这条说明了，在vue的文档下没找到:)

**Comment by janryWang** [comment]
> 看文档描述triggerType需要单独定义，可否针对组件也提供一个默认的triggerType的配置

本来就是默认的onChange触发校验的

**Reply by yuyi919** [reply]
这我明白，这里提出的是在默认配置外单独针对组件进行定义，例子就是如上提到的
虽然也可以通过decorator来修改vdom实现（不过比较难受），算是一个小建议吧
